docs/main/user-guides/replicated-storage-user-guide/replicated-pv-mayastor/rs-configuration.md

```diff
@@ -182,16 +182,259 @@ The capacity usage on a pool can be monitored using [exporter metrics](../replic
+spec:
+  node: worker-node-2
+  disks: ["/dev/sdb"]
+  topology:
+    labelled:
+        zone: us-west-1
+ ```
+
+ **Command (Get filtered pools based on labels)**
+ ```text
+kubectl mayastor get pools -n openebs --selector zone=eu-west-1
+ID             GRPC ENDPOINT        STATUS  LABELS
+ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED
+pool-on-node-0  aio:///dev/sdb?uuid=b7779970-793c-4dfa-b8d7-03d5b50a45b8  true     worker-node-0  Online  10GiB     0 B        10GiB      0 B
+pool-on-node-2  aio:///dev/sdb?uuid=b7779970-793c-4dfa-b8d7-03d5b50a45b8  true     worker-node-2  Online  10GiB     0 B        10GiB      0 B
+kubectl mayastor get pools -n openebs --selector zone=eu-east-1
+ID             GRPC ENDPOINT        STATUS  LABELS
+ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED
+pool-on-node-1  aio:///dev/sdb?uuid=b7779970-793c-4dfa-b8d7-03d5b50a45b8  true     worker-node-1  Online  10GiB     0 B        10GiB      0 B
+```
+### "PoolHasTopologyKey"
+
+The parameter 'PoolHasTopologyKey' will allow the placement of replicas on the pool that has label keys same as the keys passed in the storage class.
+
+**Command**
+```text
+cat <<EOF | kubectl create -f -
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: mayastor-1
+parameters:
+  ioTimeout: "30"
+  protocol: nvmf
+  repl: "2"
+  poolHasTopologykey: |
+    rack
+provisioner: io.openebs.csi-mayastor
+volumeBindingMode: Immediate
+```
+
+**Command (Get filtered pools based on labels)**
+ ```text
+kubectl mayastor get pools -n openebs --selector zone=eu-west-1
+ID             GRPC ENDPOINT        STATUS  LABELS
+ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED
+pool-on-node-0  aio:///dev/sdb?uuid=b7779970-793c-4dfa-b8d7-03d5b50a45b8  true     worker-node-0  Online  10GiB     0 B        10GiB      0 B
+pool-on-node-2  aio:///dev/sdb?uuid=b7779970-793c-4dfa-b8d7-03d5b50a45b8  true     worker-node-2  Online  10GiB     0 B        10GiB      0 B
+kubectl mayastor get pools -n openebs --selector zone=eu-east-1
+ID             GRPC ENDPOINT        STATUS  LABELS
+ID              DISKS                                                     MANAGED  NODE           STATUS  CAPACITY  ALLOCATED  AVAILABLE  COMMITTED
+pool-on-node-1  aio:///dev/sdb?uuid=b7779970-793c-4dfa-b8d7-03d5b50a45b8  true     worker-node-1  Online  10GiB     0 B        10GiB      0 B
+```
+
+In this case, the replica of the volume will be placed on any of the two nodes i.e.
+- `pool-on-node-1` and `pool-on-node-2` or
+- `pool-on-node-1` and `pool-on-node-3` or
+- `pool-on-node-2` and `pool-on-node-3`
+as the storage class has `rack` as the value for `nodeHasTopoogyKey` that matches with the label key of the node.
+
```

Comment:
@sinhaashish do we not have poolSpreadTopologyKey?

Reply:
No `poolSpreadTopologyKey` is not feasible 
Example 
Even though i raised it but there is an issue with the approach, now i see
As in the test case 
```
# The labels to be applied to the pools.
###############################################################################################
#   Description           ||   Pool Name   ||         Label           ||        Node        ||
#==============================================================================================
#     "pool1" has         ||   node1pool1  ||     zone-us=us-west-1   ||     io-engine-1    ||
#       the label         ||   node2pool1  ||     zone-us=us-west-1   ||     io-engine-2    ||
#   "zone-us=us-west-1"   ||   node3pool1  ||     zone-us=us-west-1   ||     io-engine-3    ||
#==============================================================================================
#     "pool2" has        ||   node1pool2  ||     zone-ap=ap-south-1   ||     io-engine-1   ||
#      the label         ||   node2pool2  ||     zone-ap=ap-south-1  ||     io-engine-2    ||
#   "zone-ap=ap-south-1" ||   node3pool2  ||     zone-ap=ap-south-1  ||     io-engine-3    ||
#============================================================================================
#     "pool3" has         ||   node1pool3  ||     zone-eu=eu-west-3   ||     io-engine-1    ||
#      the label          ||   node2pool3  ||     zone-eu=eu-west-3   ||     io-engine-2    ||
#   "zone-eu=eu-west-3"   ||   node3pool3  ||     zone-eu=eu-west-3   ||     io-engine-3    ||
#===========================================================================================
#     "poolX" has        ||   node1poolX  ||     zone-eu=eu-west-4   ||     io-engine-1    ||
#      the label         ||               ||                         ||                    ||
#   "zone-eu=eu-west-X"  ||               ||                         ||                    ||
#===========================================================================================
```
For pool spread topology key as zone-eu :
 
In a 2 replica volume, if the first replica is placed on node1 or node2 on pool3 where pool label is zone-eu=eu-west-3 the second replica goes on node3  which has the label zone-eu=eu-west-X which is correct and the test case passes.
But if the first replica sits on node3pool3 which has label `zone-eu=eu-west-3the other replica  doesn't have the pool to get provisioned because poolX is on node3 itself and the test case fails
 
There is a discrepancy .
This is for poolSpreadTopologyKey , but wont it happen for nodeSpreadTopolofyKey as the replica will pe placed on different nodes only.